src/VocabularyReferenceFieldsManager.php

```diff
@@ -0,0 +1,250 @@
+class VocabularyReferenceFieldsManager implements ContainerInjectionInterface {
+
+  /**
+   * The entity field manager.
+   *
+   * @var \Drupal\Core\Entity\EntityFieldManagerInterface
+   */
+  protected $entityFieldManager;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The reference handler plugin manager.
+   *
+   * @var \Drupal\open_vocabularies\VocabularyReferenceHandlerPluginManagerInterface
+   */
+  protected $referenceHandlerManager;
+
+  /**
+   * Creates a new instance of the class.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Drupal\Core\Entity\EntityFieldManagerInterface $entity_field_manager
+   *   The entity field manager.
+   * @param \Drupal\open_vocabularies\VocabularyReferenceHandlerPluginManagerInterface $reference_handler_manager
+   *   The vocabulary reference handler plugin manager.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, EntityFieldManagerInterface $entity_field_manager, VocabularyReferenceHandlerPluginManagerInterface $reference_handler_manager) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->entityFieldManager = $entity_field_manager;
+    $this->referenceHandlerManager = $reference_handler_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity_type.manager'),
+      $container->get('entity_field.manager'),
+      $container->get('plugin.manager.open_vocabularies.vocabulary_reference_handler')
+    );
+  }
+
+  /**
+   * Alters entity form displays to place the computed entity reference fields.
+   *
+   * @param \Drupal\Core\Entity\Display\EntityFormDisplayInterface $form_display
+   *   The entity_form_display object that will be used to display the entity
+   *   form components.
+   * @param array $context
+   *   An associative array containing:
+   *   - entity_type: The entity type, e.g., 'node' or 'user'.
+   *   - bundle: The bundle, e.g., 'page' or 'article'.
+   *   - form_mode: The form mode; e.g., 'default', 'profile', 'register', etc.
+   */
+  public function entityFormDisplayAlter(EntityFormDisplayInterface $form_display, array $context): void {
+    $association_storage = $this->entityTypeManager->getStorage('open_vocabulary_association');
+
+    $fields = $this->entityFieldManager->getFieldDefinitions($form_display->getTargetEntityTypeId(), $form_display->getTargetBundle());
+    $fields_to_hide = [];
+    foreach ($fields as $field_name => $definition) {
+      // Filter out fields that are not entity references provided by this
+      // module as base fields.
+      if (!$definition instanceof BaseFieldDefinition || $definition->getProvider() !== 'open_vocabularies' || $definition->getType() !== 'entity_reference') {
```

Comment:
I just wanted to make sure that only base fields we declared are processed here. It shouldn't process any other field declared in any other way that has that setting.

Reply:
Mmm, well, we are the ones setting that setting, so we can assume it's not there. Otherwise, any other field can just manually set that field to break stuff..

Reply:
I am conflicted, because like this we make sure that no other fields are processed aside the ones we declared with a specific scope. It also prevent extensibility on this side.